### PR TITLE
CLN: Use sentinel values instead of mutable default arguments

### PR DESCRIPTION
**Pull request by proinsias**
From [Quantified Code](https://www.quantifiedcode.com/):

Mutable list or dictionary used as default argument to a method or function. Python creates a single persistent object and uses it for every subsequent call in which the argument is left empty. This can cause problems if the program was expecting the function to return a new list or dictionary after every call. Use a sentinel value to denote an empty list or dictionary.

### CODE REVIEW

File: pandas/stats/moments.py [hunk 1]
```diff
@@ -355,7 +355,7 @@ def rolling_corr_pairwise(df1, df2=None, window=None, min_periods=None,
```
Comment (jreback) [inline pandas/stats/moments.py]: don't change code in here, this entire module is being refactored

File: pandas/tests/test_expressions.py [hunk 2]
```diff
@@ -102,13 +102,15 @@ def test_integer_arithmetic(self):
 
     @nose.tools.nottest
     def run_binary_test(self, df, other, assert_func,
-                        test_flex=False, numexpr_ops=set(['gt', 'lt', 'ge',
-                                                          'le', 'eq', 'ne'])):
```
Comment (jreback) [inline pandas/tests/test_expressions.py]: dont change this

File: pandas/tests/test_msgpack/test_pack.py [hunk 3]
```diff
@@ -84,7 +84,9 @@ def testPackFloat(self):
         assert packb(1.0, use_single_float=True)  == b'\xca' + struct.pack('>f', 1.0)
         assert packb(1.0, use_single_float=False) == b'\xcb' + struct.pack('>d', 1.0)
 
-    def testArraySize(self, sizes=[0, 5, 50, 1000]):
+    def testArraySize(self, sizes=None):
```
Comment (jreback) [inline pandas/tests/test_msgpack/test_pack.py]: don't change this (or make it a tuple)

File: pandas/tests/test_msgpack/test_pack.py [hunk 4]
```diff
@@ -97,7 +99,9 @@ def testArraySize(self, sizes=[0, 5, 50, 1000]):
         for size in sizes:
             assert unpacker.unpack() == list(range(size))
 
-    def test_manualreset(self, sizes=[0, 5, 50, 1000]):
+    def test_manualreset(self, sizes=None):
```
Comment (jreback) [inline pandas/tests/test_msgpack/test_pack.py]: same

File: doc/sphinxext/numpydoc/docscrape.py [hunk 5]
```diff
@@ -86,7 +86,9 @@ def is_empty(self):
 
 
 class NumpyDocString(object):
-    def __init__(self, docstring, config={}):
```
Comment (jreback) [inline doc/sphinxext/numpydoc/docscrape.py]: don't change anything in the doc building, these are pulled in external modules

**Comment by jreback** [comment]
ok, but really need to limit scope to tested code, and need to be really careful changing the tests

**Comment by proinsias** [comment]
Should be all set now.

**Comment by jreback** [comment]
this is fine. ping when green.

**Comment by proinsias** [comment]
All checks have passed.

**Comment by jreback** [comment]
thanks!

**Pull request merged**